packages/components/emoji/src/emoji-browser.ts

```diff
@@ -0,0 +1,246 @@
+  @property({ attribute: 'frequently-used' }) frequentlyUsed?: string;
+
+  /** @internals The frequently used emojis. */
+  @state() frequentlyUsedEmojis: Emoji[] = [];
+
+  /** @internal The emojis, grouped by group. */
+  @state() emojis?: Map<EmojiGroup, Emoji[]>;
+
+  /**
+   * The locale for this component.
+   * TODO: Use the LocaleMixin.
+   */
+  @property() locale?: string;
+
+  /** The query string to filter emojis. */
+  @property() query?: string;
+
+  /** @internal Emits when the user selects an emoji. */
+  @event({ name: 'sl-select' }) selectEvent!: EventEmitter<SlSelectEvent<Emoji>>;
+
+  /** @internal Service for getting the data. */
+  service?: EmojiService;
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    if (this.baseUrl) {
+      this.service = new EmojiService(this.baseUrl);
+    } else {
+      throw new Error('The `baseUrl` property is required when using the sl-emoji-browser component.');
+    }
+  }
+
+  override async willUpdate(changes: PropertyValues<this>): Promise<void> {
+    super.willUpdate(changes);
+
+    if (changes.has('frequentlyUsed')) {
+      const emojis = await this.service?.getEmojis(this.locale as SupportedLocale);
+
+      this.frequentlyUsedEmojis = this.frequentlyUsed
+        ?.split(' ')
+        .map(unicode => emojis?.find(emoji => emoji.unicode === unicode))
+        .filter(Boolean) as Emoji[];
+    }
+
+    if (changes.has('locale')) {
+      this.emojis = await this.service?.getGroupedEmojis(this.locale as SupportedLocale);
+    }
+
+    if (changes.has('query')) {
+      if (this.query) {
+        this.filteredEmojis = (await this.service?.findEmojis(this.locale as SupportedLocale, this.query)) ?? [];
+      } else {
+        this.filteredEmojis = [];
+      }
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <sl-tab-group>
+        ${this.frequentlyUsedEmojis?.length
+          ? html`
+              <sl-tab @click=${this.#onTabClick} id="group-frequently-used">
+                <sl-icon name="far-clock" slot="icon"></sl-icon>
+              </sl-tab>
+            `
+          : nothing}
+        ${map(
+          this.emojis?.keys(),
+          group => html`
+            <sl-tab @click=${this.#onTabClick} .id=${`group-${group.key}`}>
+              <sl-icon .name=${GROUP_ICONS[group.order]} slot="icon"></sl-icon>
+            </sl-tab>
+          `
+        )}
+      </sl-tab-group>
+
+      <div @scroll=${this.#onScroll} part="wrapper">
+        <sl-search-field
+          @sl-change=${this.#onChange}
+          @sl-clear=${this.#onClear}
+          .placeholder=${msg('Search')}
+          .value=${this.query}
+        ></sl-search-field>
+
+        ${this.filteredEmojis.length
+          ? this.renderEmojis(this.filteredEmojis)
+          : html`
+              ${this.frequentlyUsedEmojis?.length
+                ? html`
+                    <h1 id="frequently-used">${msg('Frequently Used')}</h1>
```

Comment:
> Maybe it's better to use sth else than `h1` here? Maybe it will not be a really good idea to have multiple `h1` elements on a single page for people who will use screen readers?

I'll ask around if any knows of any best practices.

Reply:
For modal screens it's accepted to use h1, here i would not recommend it
https://medium.com/@web-accessibility-education/which-heading-level-should-dialog-modals-have-7b3df89437f0
https://github.com/mui/material-ui/issues/34250

Reply:
Changed to use a div instead of h1.